turbinia/evidence.py

```diff
@@ -524,34 +528,58 @@ class DiskPartition(RawDisk):
   """Evidence object for a partition within Disk based evidence.
 
   Attributes:
-    path_spec (dfvfs.PathSpec): Partition path spec.
+    partition_location: dfVFS partition location.
+    type_indicator: dfVFS path_spec type indicator.
     partition_offset: Offset of the partition in bytes.
     partition_size: Size of the partition in bytes.
+    path_spec (dfvfs.PathSpec): Partition path spec.
   """
 
   POSSIBLE_STATES = [EvidenceState.ATTACHED, EvidenceState.MOUNTED]
 
   def __init__(
-      self, path_spec=None, partition_offset=None, partition_size=None, *args,
-      **kwargs):
+      self, partition_location=None, partition_offset=None, partition_size=None,
+      path_spec=None, *args, **kwargs):
     """Initialization for raw volume evidence object."""
 
-    self.path_spec = path_spec
+    self.partition_location = partition_location
     self.partition_offset = partition_offset
     self.partition_size = partition_size
+    self.path_spec = path_spec
     super(DiskPartition, self).__init__(*args, **kwargs)
 
     # This Evidence needs to have a parent
     self.context_dependent = True
 
   def _preprocess(self, _, required_states):
+    # We need to enumerate partitions in preprocessing so the path_specs match
+    # the parent evidence location for each task.
+    try:
+      path_specs = partitions.Enumerate(self.parent_evidence)
+    except Error as e:
```

Comment:
Looking at `Enumerate()` it looks like it will raise a  dfvfs_errors.ScannerError, should we trap that?   Or maybe we should have that method raise `TurbiniaException` instead so that we don't have to import dfvfs here.